### ISSUE TITLE
Solution to issues with Relevansii + wp-advanced-search + custom post types 

### ISSUE THREAD

**Issue by deweydb** [bootsz/wp-advanced-search#173]
I saw a lot of other issues on this tracker mentioning the same thing.  I think i finally found the cause.

https://github.com/bootsz/wp-advanced-search/blob/master/src/Factory.php#L193

the line that defines $query_args is commented out, but then 2 lines later you attempt to use this variable.  This results in all queries being defaulted to search for post_type = 'post'; which will return zero results if the user is attempting to search custom post types.

My fix was to modify this line:

$query->query_vars['post_type'] = (empty($query_args['post_type'])) ? 'post' : $query_args['post_type'];

to this:

$query->query_vars['post_type'] = (empty($this->wp_query_args['post_type'])) ? 'post' : $this->wp_query_args['post_type'];

But i'm not sure if this is the _best_ fix.

Thanks for the plugin, it is awesome!

**Comment by bootsz** [comment]
Ah, good catch!  That part of the code was changed recently, I clearly was not paying close enough attention.  This fixed the issue for you?

I'm not sure that this is the answer to all of the integration problems with Relevanssi, as there have been issues with it for a long time (even before this specific line was changed), but it should at least fix this particular problem.  Thanks for looking into it, I will try to push an update soon.

**Comment by deweydb** [comment]
Yeah, that fixed it for me. and definitely not the cause of all the integration issues in this tracker, but i'm pretty sure the cause of #155

**Comment by Mte90** [comment]
I've tried for fix https://github.com/bootsz/wp-advanced-search/issues/176 but doesn't work

**Comment by Mte90** [comment]
I think that also that line is useless because the wp_query is already executed

**Comment by deweydb** [comment]
@Mte90 no. that is definitely 100% an incorrect statement.  Add some file_put_contents() around there and see what happens for yourself.

**Comment by Mte90** [comment]
As I can see the WP_query have a different query_vars['post_type'] but $query->getArgs() have the correct post_type. 
As I can see that project don't use many actions or filters but I don't understand how is changing.

**Comment by deweydb** [comment]
Yeah, it has to do with where Relevansii looks for those post_type.  It is set in two places (for some reason that is beyond my comprehension of wordpress).  And without the above line it is only set in the wrong place, not the place that Relevansii looks for post_types.
